### ISSUE TITLE
Curly bugs

### ISSUE THREAD

**Issue by Mister-Hope** [eslint/eslint#13280]
**Tell us about your environment**

* **ESLint Version:**
7.0.0
* **Node Version:**
12.16.3
* **npm Version:**
yarn 1.24.4

**What parser (default, Babel-ESLint, etc.) are you using?**
eslint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
rules:
  curly:
    - error
    - multi
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  for (const [lineId, line] of lines.entries())
    if (regexp === false) {
      for (const reg of regionRegexps)
        if (testLine(line, reg, regionName)) {
          start = lineId + 1;
          regexp = reg;
          break;
        }
    } else if (testLine(line, regexp, regionName, true))
      return { start, end: lineId, regexp };
```

```bash
eslint .
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```json
{
	"resource": "/z:/npm/hopepress/packages/@vuepress/markdown/lib/snippet.ts",
	"owner": "eslint",
	"code": {
		"value": "curly",
		"target": {
			"$mid": 1,
			"external": "https://eslint.org/docs/rules/curly",
			"path": "/docs/rules/curly",
			"scheme": "https",
			"authority": "eslint.org"
		}
	},
	"severity": 8,
	"message": "Unnecessary { after 'if' condition.",
	"source": "eslint",
	"startLineNumber": 2,
	"startColumn": 3,
	"endLineNumber": 2,
	"endColumn": 3
}
```

The most worse part is I am using `eslint --fix` earlier, and the code became

```js
  for (const [lineId, line] of lines.entries())
    if (regexp === false) 
      for (const reg of regionRegexps)
        if (testLine(line, reg, regionName)) {
          start = lineId + 1;
          regexp = reg;
          break;
        }
       else if (testLine(line, regexp, regionName, true))
        return { start, end: lineId, regexp };
```

which is tolally different.

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by anikethsaha** [comment]
Thanks for the issue. 

This is the intended behavior as you are using `multi` option for the rule and it allows `{ }` only if it contains more than one statement in them. 

so in your code. 

```js
  for (const [lineId, line] of lines.entries())  // single statement - ok
    if (regexp === false) {  // single statement - eslint will throw error as this is single statement but you have placed them in blocks i.e wrapped with { ... }
      for (const reg of regionRegexps)  // single statement - ok
        if (testLine(line, reg, regionName)) {  // multi statement - ok
          start = lineId + 1;
          regexp = reg;
          break;
        }
    } else if (testLine(line, regexp, regionName, true)) 
      return { start, end: lineId, regexp };

```


[refer this](https://eslint.org/docs/rules/curly#multi)

**Comment by mdjermanovic** [comment]
@Mister-Hope thanks for the bug report!

I can't reproduce this in our [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGN1cmx5OiBbXCJlcnJvclwiLCBcIm11bHRpXCJdICovXG5cbiAgZm9yIChjb25zdCBbbGluZUlkLCBsaW5lXSBvZiBsaW5lcy5lbnRyaWVzKCkpXG4gICAgaWYgKHJlZ2V4cCA9PT0gZmFsc2UpIHtcbiAgICAgIGZvciAoY29uc3QgcmVnIG9mIHJlZ2lvblJlZ2V4cHMpXG4gICAgICAgIGlmICh0ZXN0TGluZShsaW5lLCByZWcsIHJlZ2lvbk5hbWUpKSB7XG4gICAgICAgICAgc3RhcnQgPSBsaW5lSWQgKyAxO1xuICAgICAgICAgIHJlZ2V4cCA9IHJlZztcbiAgICAgICAgICBicmVhaztcbiAgICAgICAgfVxuICAgIH0gZWxzZSBpZiAodGVzdExpbmUobGluZSwgcmVnZXhwLCByZWdpb25OYW1lLCB0cnVlKSlcbiAgICAgIHJldHVybiB7IHN0YXJ0LCBlbmQ6IGxpbmVJZCwgcmVnZXhwIH07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7Imdsb2JhbFJldHVybiI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6e319fQ==) (v7.0.0).

Please let me know if I'm missing something.

I can confirm that this bug indeed exists in v.6.8.0, but it should have been fixed by #12943, which was merged for v7.0.0.

**Comment by Mister-Hope** [comment]
emm I do have a version 6.8.0 globally installed, I will check jt out after I return home. thanks
